search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,278 @@
+    var set = 0;
+    // 100 queries per minute limit for Custom Search API. Pause to prevent
+    // surpassing limit.
+    if (i !== 0 && i % 100 === 0) {
+      await new Promise(resolve => setTimeout(resolve, 60000));
+    }
+    let countryData = [];
+    // Update countryData within the functions called.
+    const avg = await getSearchResultsForCountryFromAPI(
+      "country" + json[i].id, query, countryData);
+    console.log('ntarn debug country: ' + json[i].id + ' averageSentiment: ' + avg);
+    countriesData.push({
+      country: json[i].id,
+      averageSentiment: avg,
+      results: countryData,
+    });
+  }
+  addTopicToDatastore(query, countriesData);
+}
+
+/** 
+ * Gets top 10 search results for specified country and query from Custom
+ * Search API and passes them to be saved in countryData object.
+ * @param {string} countryCode 2 letter country code for search results to be
+ *     written in.
+ * @param {string} query Search query.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function getSearchResultsForCountryFromAPI(countryCode, query, countryData) {
+  let response =
+    await fetch('https://www.googleapis.com/customsearch/v1?key=AIzaSyDszWv1aGP7Q1uOt74CqBpx87KpkhDR6Io&cx=017187910465527070415:o5pur9drtw0&q=' + query + '&cr=' + countryCode + '&num=10&safe=active&dateRestrict=d1&fields=items(title,snippet,htmlTitle,link)');
+  let searchResults = await response.json();
+  return await saveResultsAndDeletePrevious(searchResults, countryData);
+}
+
+/**
+ * Deletes previous search results and saves the current results in the
+ * Datastore, given the results JSON obtained from the API.
+ * @param {Object} searchResultsJson Object with information for top 10 search
+ *     results.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function saveResultsAndDeletePrevious(searchResultsJson, countryData) {
+  await deleteAncientResults();
+
+  // Parse the JSON string and pass each search result to add to the
+  // countryData object.
+  var currentSearchResults = searchResultsJson.items;
+  try {
+    let avg = 0;
+    if (currentSearchResults == undefined) {
+      return 0;
+    } else {
+      for (var i = 0; i < currentSearchResults.length; i++) {
+        avg = avg + await addSearchResultToCountryData(currentSearchResults[i], countryData);
+        console.log('avg in for loop of saveResults' + avg);
+      }
+      return avg / currentSearchResults.length;
+    }
+  } catch (err) { // Occurs when no search results for that country and topic.
+    console.error('ERROR:', err);
+    countryData = null;
+  }
+}
+
+/** Deletes search results from 7 days ago. */
+async function deleteAncientResults() {
+  const query = datastore.createQuery('CustomSearchTopic').order('timestamp');
+  const [searchResults] = await datastore.runQuery(query);
+
+  // Note: Can't use forEach with await.
```

Comment:
Remember and remove this note

Reply:
I think it's best if we delete it further on or when we deploy, because all of us forgot at some point in time last week. @lawrencezjh @hayleejane3 What do you guys think? Understand if we should delete it but it caused some promise warnings when we forgot about this

Reply:
Add a todo if this is meant to be deleted.